fecgov/fec-eregs
# update dependencies

## Summary (required)

Update node/npm to v14.15.4/8.3.1. Some dev dependencies require node 14.15.x or higher. Else getting NPM warning with unsupported engine errors shown in the screenshot below.

1. Update following in package.json 

- node-sass v7.0.1
- lodash v4.17.21

2. Generate package-lock.json with dependencies (form latest package.json file changes)

-  json-schema v0.4.0
-  ansi-regex -v5.0.1

3. Update node in circle/config.yml

- Resolves #645 #646


### Required reviewers

Two developer 

## Screenshots

run `snyk test --file=package.json` on `develop` branch:
**Before:**
<img width="1217" alt="Screen Shot 2022-01-19 at 8 33 22 PM" src="https://user-images.githubusercontent.com/11650355/151087975-086f65cd-f739-45c3-8426-27ab0d1b2783.png">


After:
<img width="1101" alt="Screen Shot 2022-01-19 at 8 33 08 PM" src="https://user-images.githubusercontent.com/11650355/151087899-d36340cb-0307-4f54-abf2-a5866e4d73f2.png">

run `snyk test --file=package.json` on `feature` branch:
<img width="1218" alt="Screen Shot 2022-01-19 at 9 26 14 PM" src="https://user-images.githubusercontent.com/11650355/151087929-c6c57c48-323f-4502-8839-6725f0fb8ff6.png">


## How to test

- checkout branch
- pyenv activate venv-eregs-3712 (install requirements.txt in case you haven't)
- rm -rf node_modules/
- nvm install v14.15.5
- nvm alias default v14.15.5
- nvm use default
- npm install -g npm@8.3.1 (npm@8.3.2 is latest)
- npm i
- npm run build
- python manage.py migrate
- python manage.py compile_frontend
- npm install -g grunt-cli (In case you run into `No such file or directory: 'grunt': 'grunt'`)
- run `snyk test --file=package.json` ( json-schema, ansi-regex, lodash vulnerabilities no longer show except node-sass v7.0.1 with 9 vulnerabilities)
- python manage.py runserver (make sure the 45 regs display OK on the browser)
